### PR TITLE
Fix kubelet image GC by using new `image_id` CRI field

### LINKED ISSUE #7143: imageRef in container status is not image ID and lead to kubelet image gc work abnormally
### What happened?

The imageRef of containers, returned by runtime cri-o, may not be image ID. This will mislead kubelet's image-gc logical.

In [CRI spec](https://github.com/kubernetes/kubernetes/blob/90c362b3430bcbbf8f245fadbcd521dab39f1d7c/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.pb.go#L5544), it consider imageRef as image ID:
```
// Reference to the image in use. For most runtimes, this should be an
// image ID.
ImageRef string `protobuf:"bytes,5,opt,name=image_ref,json=imageRef,proto3" json:"image_ref,omitempty"`
```

When do image gc logical, kubelet will get image and container status from cri-o runtime service. And then, compare currently used images (by container) and existing images by image ID.

If imageRef return by cri-o is not image ID, kubelet image gc can not work properly.

Kubelet's key logs may look like this:
```
I0719 16:36:14.498634 xxx image_gc_manager.go:233] "Container uses image" pod="multus/multus-admission-controller-7ff9c677b7-2p6lx" containerName="multus-admission-controller" containerImage="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" imageID="image.foo.bar/multus-admission-controller@sha256:36f44236d5c1aa57bd5e930179f020c089e2bde0d20a5d349f5af8fefe2d6235"
I0719 16:36:14.500180 xxx image_gc_manager.go:244] "Adding image ID to currentImages" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
I0719 16:36:14.500186 xxx image_gc_manager.go:261] "Image ID has size" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" size=316580906
I0719 16:36:14.500192 xxx image_gc_manager.go:264] "Image ID is pinned" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" pinned=false
I0719 16:36:14.534022 xxx image_gc_manager.go:371] "Evaluating image ID for possible garbage collection" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
I0719 16:36:14.534031 xxx image_gc_manager.go:387] "Removing image to free bytes" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" size=316580906
E0719 16:36:14.534609 xxx remote_image.go:270] "RemoveImage from image service failed" err="rpc error: code = Unknown desc = Image used by c5ea50c909f1794ec8cf7e10e3e740bbd76f4d27d8c8af7625f84ef844868128: image is in use by a container" image="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
E0719 16:36:14.534633 xxx kuberuntime_image.go:122] "Failed to remove image" err="rpc error: code = Unknown desc = Image used by c5ea50c909f1794ec8cf7e10e3e740bbd76f4d27d8c8af7625f84ef844868128: image is in use by a container" image="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
```

The real image ID is *38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3*, and image name is *image.foo.bar/multus-admission-controller@sha256:36f44236d5c1aa57bd5e930179f020c089e2bde0d20a5d349f5af8fefe2d6235*. However, kubelet just get the opposite way:
```
I0719 16:36:14.498634 xxx image_gc_manager.go:233] "Container uses image" ... containerImage="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" imageID="image.foo.bar/multus-admission-controller@sha256:36f44236d5c1aa57bd5e930179f020c089e2bde0d20a5d349f5af8fefe2d6235"
```

### What did you expect to happen?

imageRef returned by cri-o in container status should be image ID, and be consistent with image id information return by runtime image service.

### How can we reproduce it (as minimally and precisely as possible)?

Making a kubernetes node under disk pressure, and trigger kubelet image gc.

Image gc would continously try to remove image, which is used by container, and failed. The kubelet logs (log level 5) may continously output like this:
```
I0719 16:36:14.498634 xxx image_gc_manager.go:233] "Container uses image" pod="multus/multus-admission-controller-7ff9c677b7-2p6lx" containerName="multus-admission-controller" containerImage="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" imageID="image.foo.bar/multus-admission-controller@sha256:36f44236d5c1aa57bd5e930179f020c089e2bde0d20a5d349f5af8fefe2d6235"
I0719 16:36:14.500180 xxx image_gc_manager.go:244] "Adding image ID to currentImages" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
I0719 16:36:14.500186 xxx image_gc_manager.go:261] "Image ID has size" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" size=316580906
I0719 16:36:14.500192 xxx image_gc_manager.go:264] "Image ID is pinned" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" pinned=false
I0719 16:36:14.534022 xxx image_gc_manager.go:371] "Evaluating image ID for possible garbage collection" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
I0719 16:36:14.534031 xxx image_gc_manager.go:387] "Removing image to free bytes" imageID="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3" size=316580906
E0719 16:36:14.534609 xxx remote_image.go:270] "RemoveImage from image service failed" err="rpc error: code = Unknown desc = Image used by c5ea50c909f1794ec8cf7e10e3e740bbd76f4d27d8c8af7625f84ef844868128: image is in use by a container" image="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
E0719 16:36:14.534633 xxx kuberuntime_image.go:122] "Failed to remove image" err="rpc error: code = Unknown desc = Image used by c5ea50c909f1794ec8cf7e10e3e740bbd76f4d27d8c8af7625f84ef844868128: image is in use by a container" image="38da782fd6a202da62c652532b5d289278819d84bf2f7280ad40d214903db6e3"
```

### Anything else we need to know?

_No response_

### CRI-O and Kubernetes version

<details>

```console
$ crio --version
crio version 1.24.3
Version:          1.24.3
GitCommit:        0e724229f2a81e08a5e05cef0b125ee0dadda08e
```

kubernetes is compile by ourself, which is forked from *https://github.com/kubernetes/kubernetes/tree/v1.25.8*

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here
```

</details>

### Additional environment details (AWS, VirtualBox, physical, etc.)

<details>
VirtualBox
</details>

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Using the implementation of https://github.com/kubernetes/kubernetes/pull/123508 to fix https://github.com/cri-o/cri-o/issues/7143

Also incorporating https://github.com/kubernetes/kubernetes/pull/123583 now.

#### Which issue(s) this PR fixes:

Fixes #7143

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Fixed kubelet image garbage collection when images being referenced as digests instead of IDs. 
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Waiting for the [staging repositories](https://github.com/kubernetes/api/commits/master/) to pick-up commit [e663285c](https://github.com/kubernetes/kubernetes/commit/e663285ccf4097456921918bd1888ba544f79cef).

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -84,12 +84,15 @@ jobs:
...
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
       - uses: actions/download-artifact@v4
         with:
           name: build
           path: bin
       - run: |
-          sudo -E make check-nri-bats-tests
+          sudo -E make check-nri-bats-tests GOSUMDB=
```
Comment (saschagrunert) [inline .github/workflows/test.yml:95]: Interesting issue we also have in prow CI:

```
verifying module: invalid GOSUMDB: malformed verifier id
```

**Comment by haircommander** [comment]
@saschagrunert do we still want this? can you rebase?

**Comment by saschagrunert** [comment]
Rebased on top of the latest `main` branch.

**Comment by sohankunkerkar** [comment]
Closing this in favor of https://github.com/cri-o/cri-o/pull/8115

**Pull request closed without merging**